src/aiida/orm/querybuilder.py

```diff
@@ -1105,7 +1105,13 @@ def iterdict(self, batch_size: Optional[int] = 100) -> Iterable[Dict[str, Dict[s
 
             yield item
 
-    def all(self, batch_size: Optional[int] = None, flat: bool = False) -> Union[List[List[Any]], List[Any]]:
+    @overload
+    def all(self, batch_size: int | None = None, flat: Literal[False] = False) -> list[list[Any]]: ...
```

Comment:
I'm not sure, but I think the default value here is redundant 
```suggestion
    def all(self, batch_size: int | None = None, flat: Literal[False]) -> list[list[Any]]: ...
```

Reply:
Unfortunately, without a default  value I get a syntax error because the previous argument has a default value.

Reply:
Yeah, that would make sense, but unfortunately mypy complains

> src/aiida/orm/querybuilder.py:1112: error: Incompatible default for argument "flat" (default has type "Literal[False]", argument has type "Literal[True]")  [assignment]

I am not sure if there's a different solution here, I don't have experience with `@overload` decorator. Maybe @unkcpz or @sphuber have ideas?